cf-oscore/src/test/java/org/eclipse/californium/oscore/EndpointContextInfoTest.java

```diff
@@ -0,0 +1,406 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+
+/**
+ * Tests setting and checking of OSCORE information in the
+ * source/destination endpoint context during communication.
+ *
+ */
+@Category(Medium.class)
+public class EndpointContextInfoTest {
+	@ClassRule
+	public static CoapNetworkRule network = new CoapNetworkRule(CoapNetworkRule.Mode.DIRECT, CoapNetworkRule.Mode.NATIVE);
+
+	private static String SERVER_RESPONSE = "server responds hi";
+
+	private CoapServer server;
+
+	private Endpoint serverEndpoint;
+
+	//OSCORE context information for server and client
+	private final static HashMapCtxDB dbServer = new HashMapCtxDB();
+	private final static HashMapCtxDB dbClient = new HashMapCtxDB();
+	private final static AlgorithmID alg = AlgorithmID.AES_CCM_16_64_128;
+	private final static AlgorithmID kdf = AlgorithmID.HKDF_HMAC_SHA_256;
+	private final static byte[] master_secret = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B,
+			0x0C, 0x0D, 0x0E, 0x0F, 0x10 };
+	private final static byte[] context_id = { 0x74, 0x65, 0x73, 0x74 };
+
+	//Keys used in the endpoint context map of strings
+	private final static String OSCORE = OSCoreEndpointContextInfo.OSCORE;
+	private final static String OSCORE_SENDER_ID = OSCoreEndpointContextInfo.OSCORE_SENDER_ID;
+	private final static String OSCORE_RECIPIENT_ID = OSCoreEndpointContextInfo.OSCORE_RECIPIENT_ID;
+	private final static String OSCORE_CONTEXT_ID = OSCoreEndpointContextInfo.OSCORE_CONTEXT_ID;
+	private final static String OSCORE_URI = OSCoreEndpointContextInfo.OSCORE_URI;
+
+	@Before
+	public void initLogger() {
+		System.out.println(System.lineSeparator() + "Start " + getClass().getSimpleName());
+		EndpointManager.clear();
+	}
+
+	//Use the OSCORE stack factory
+	@BeforeClass
+	public static void setStackFactory() {
+		OSCoreCoapStackFactory.useAsDefault(dbClient);
+	}
+
+	@After
+	public void after() {
+		if (null != server) {
+			server.destroy();
+		}
+		System.out.println("End " + getClass().getSimpleName());
+	}
+
+	/**
+	 * Sends two requests and checks that appropriate OSCORE information has
+	 * been set in the following:
+	 * Server: Request source context & Response destination context
+	 * Client: Response source context
+	 *
+	 * @throws Exception if message processing or creating the OSCORE context fails
+	 */
+	@Test
+	public void testEndpointContextInfo() throws Exception {
+	    createSimpleServer();
+
+	    //Set up OSCORE context information for request (client)
+	    byte[] sidClient = new byte[] { 0x77, 0x66, 0x55, 0x44 };
+	    byte[] ridClient = new byte[] { 0x01, 0x02, 0x03, 0x04 };
+	    OSCoreCtx ctx = new OSCoreCtx(master_secret, true, alg, sidClient, ridClient, kdf, 32, null, context_id);
+	    String serverUri = "coap://" + serverEndpoint.getAddress().getAddress().getHostAddress() + ":" + serverEndpoint.getAddress().getPort();
```

Comment:
Did you check, if 
`serverEndpoint.getUri().toASCIIString()`
works for you?

Reply:
Yes, thanks for the tip. Changed in commit 104bb408c84bd45b35980d0454472a833ec4dff5